Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -273,40 +273,25 @@ public class ByteBuffer: Codable {
     }
 }
 
-extension ByteBuffer: AwsStream {
-    public var status: aws_stream_status {
-        return aws_stream_status(is_end_of_stream: self.currentIndex == array.count, is_valid: true)
-    }
+extension ByteBuffer: IStreamable {
 
-    public var length: UInt {
-        return UInt(array.count)
+    public func length() throws -> UInt64 {
+        return UInt64(array.count)
     }
 
-    public func seek(offset: Int64, basis: aws_stream_seek_basis) -> Bool {
-        let targetOffset: Int64
-        if basis.rawValue == AWS_SSB_BEGIN.rawValue {
-            targetOffset = offset
-
-        } else {
-            targetOffset = Int64(length) - offset
-        }
-        currentIndex = Int(targetOffset)
-        return true
+    public func seek(offset: UInt64) throws {
+        currentIndex = Int(offset)
```

Comment:
check that currentIndex doesn't exceed capacity

(oh yeah, just remembered, you were doing that safety check in the wrapper function, but I don't think you should, for efficiency's sake. Sorry about getting so hung up on efficiency on the aws_input_stream implementation, but this is is literally where our programs spend most of their time)